src/interfaces/assistants_web/src/hooks/session.ts

```diff
@@ -55,15 +51,18 @@ export const useSession = () => {
 
   const loginMutation = useMutation<JWTResponse | null, ApiError, LoginParams>({
     mutationFn: (params) => cohereClient.login(params),
-    onSuccess: (data: JWTResponse | null) => {
-      setAuthToken(data?.token);
+    onSuccess: async (data: JWTResponse | null) => {
+      if (!data) {
+        return new Promise((_, reject) => reject(new Error('Invalid login')));
+      }
+      await setAuthToken(data.token);
       return new Promise((resolve) => resolve(data?.token));
```

Comment:
With this function now being async, I think we can change the `return` statement to just be
```
return data?.token
```

(This is also true for the other 2 functions here that return Promise-wrapped values)